builtin/providers/google/resource_container_cluster_test.go

```diff
@@ -135,53 +133,124 @@ func testAccCheckContainerClusterDestroy(s *terraform.State) error {
+			{"initial_node_count", strconv.FormatInt(cluster.InitialNodeCount, 10)},
+			{"master_auth.0.client_certificate", cluster.MasterAuth.ClientCertificate},
+			{"master_auth.0.client_key", cluster.MasterAuth.ClientKey},
+			{"master_auth.0.cluster_ca_certificate", cluster.MasterAuth.ClusterCaCertificate},
+			{"master_auth.0.password", cluster.MasterAuth.Password},
+			{"master_auth.0.username", cluster.MasterAuth.Username},
+			{"zone", cluster.Zone},
+			{"cluster_ipv4_cidr", cluster.ClusterIpv4Cidr},
+			{"description", cluster.Description},
+			{"endpoint", cluster.Endpoint},
+			{"instance_group_urls", cluster.InstanceGroupUrls},
+			{"logging_service", cluster.LoggingService},
+			{"monitoring_service", cluster.MonitoringService},
+			// TODO(danawillow): Add this back in. Currently this field is saved via the config instead of from the API response,
+			// and the config may contain the network name or self_link, whereas the API only returns the self_link.
+			// {"network", cluster.Network},
+			{"subnetwork", cluster.Subnetwork},
+			{"node_config.0.machine_type", cluster.NodeConfig.MachineType},
+			{"node_config.0.disk_size_gb", strconv.FormatInt(cluster.NodeConfig.DiskSizeGb, 10)},
+			{"node_config.0.oauth_scopes", cluster.NodeConfig.OauthScopes},
+			{"node_version", cluster.CurrentNodeVersion},
 		}
 
-		additionalZonesSize, err := strconv.Atoi(rs.Primary.Attributes["additional_zones.#"])
-		if err != nil {
-			return err
+		// Remove Zone from additional_zones since that's what the resource writes in state
+		additionalZones := []string{}
+		for _, location := range cluster.Locations {
+			if location != cluster.Zone {
+				additionalZones = append(additionalZones, location)
+			}
 		}
-		if additionalZonesSize != num {
-			return fmt.Errorf("number of additional zones did not match %d, was %d", num, additionalZonesSize)
+		clusterTests = append(clusterTests, clusterTestField{"additional_zones", additionalZones})
+
+		// AddonsConfig is neither Required or Computed, so the API may return nil for it
+		if cluster.AddonsConfig != nil {
+			if cluster.AddonsConfig.HttpLoadBalancing != nil {
+				clusterTests = append(clusterTests, clusterTestField{"addons_config.0.http_load_balancing.0.disabled", strconv.FormatBool(cluster.AddonsConfig.HttpLoadBalancing.Disabled)})
+			}
+			if cluster.AddonsConfig.HorizontalPodAutoscaling != nil {
+				clusterTests = append(clusterTests, clusterTestField{"addons_config.0.horizontal_pod_autoscaling.0.disabled", strconv.FormatBool(cluster.AddonsConfig.HorizontalPodAutoscaling.Disabled)})
+			}
+		}
+
+		for _, attrs := range clusterTests {
+			if c := checkMatch(attributes, attrs.tf_attr, attrs.gcp_attr); c != "" {
+				return fmt.Errorf(c)
+			}
 		}
 
 		return nil
 	}
 }
 
+func getResourceAttributes(n string, s *terraform.State) (map[string]string, error) {
+	rs, ok := s.RootModule().Resources[n]
+	if !ok {
+		return nil, fmt.Errorf("Not found: %s", n)
+	}
+
+	if rs.Primary.ID == "" {
```

Comment:
It doesn't really have an obvious place it belongs. The flow of this section of code in the original (and in most resources) seems to be leading towards getting the Attributes, but returning an error if certain preconditions aren't met- the resource isn't there, or has an empty id. From that point on, the only part of the ResourceState that gets used is the Attributes.